backpack-common/src/main/java/net/skyscanner/backpack/calendar2/data/CalendarCell.kt

```diff
@@ -24,27 +24,32 @@ import androidx.core.text.buildSpannedString
 import net.skyscanner.backpack.calendar2.CalendarParams
 import net.skyscanner.backpack.calendar2.CalendarSelection
 import net.skyscanner.backpack.calendar2.CellInfo
+import net.skyscanner.backpack.util.InternalBackpackApi
 import org.threeten.bp.LocalDate
 import org.threeten.bp.YearMonth
 
-internal sealed class CalendarCell {
+@InternalBackpackApi
+sealed class CalendarCell {
```

Comment:
Ideally, we also want to annotate these with `@Stable/@Immutable` as Compose won't be able to calculate it itself. However, that means including Compose Runtime into the dependencies of the module. Need you opinion.

Reply:
If there's a function taking CalendarCell, only this function will be recomposed.

However, the entire calendar is a function from CalendarState. And that changes with every interaction, so it will cause recomposition for the entire calendar.

Compose can skip individual cells, but only if the items are stable or immutable. Compose can also determine in a lot of cases whether the item is immutable. Lists are NOT considered to be immutable by default. Java types (LocalDate) that do not have properties in the constructor are not considered to be immutable either.

Because these rules are known I haven't checked the actual recompositions – there's no way the compiler can consider the state to be immutable at this point.

Reply:
OK in which case: can you benchmark the performance to see if it's a problem? If it is we need to look at what we can do